framework/wazuh/rule.py

```diff
@@ -420,45 +461,57 @@ def upload_rule_file(filename: str = None, content: str = None, overwrite: bool
         result.affected_items.append(to_relative_path(full_path))
         result.total_affected_items = len(result.affected_items)
         backup_file and exists(backup_file) and remove(backup_file)
-    except WazuhError as e:
-        result.add_failed_item(id_=to_relative_path(full_path), error=e)
+    except WazuhError as exc:
+        result.add_failed_item(id_=to_relative_path(full_path), error=exc)
     finally:
         exists(backup_file) and safe_move(backup_file, full_path)
 
     return result
 
 
 @expose_resources(actions=['rules:delete'], resources=['rule:file:{filename}'])
-def delete_rule_file(filename: str = None) -> AffectedItemsWazuhResult:
+def delete_rule_file(filename: str, relative_dirname: str = None) -> AffectedItemsWazuhResult:
     """Delete a rule file.
 
     Parameters
     ----------
     filename : str, optional
-        Name of the rule file. Default `None`
+        Name of the rule file.
+    relative_dirname : str
+        Relative directory where the rule file is located.
 
     Returns
     -------
     AffectedItemsWazuhResult
         Affected items.
     """
+    file = filename if isinstance(filename, str) else filename[0]
```

Comment:
We could change `filename` type hint to `typing.Union[str, list]` because doing `isinstance(filename, str)` when the type hint is `str` is confusing.